nuxeo-core/nuxeo-core-api/src/main/java/org/nuxeo/ecm/core/api/security/ACE.java

```diff
@@ -69,11 +73,21 @@ public static ACE fromId(String aceId) {
             return null;
         }
 
-        String[] parts = aceId.split(":");
-        if (parts.length < 3) {
+        // An ACE is composed of tokens separated with ":" caracter
+        // First 3 tokens are mandatory; following 3 tokens are optional
+        // The ":" separator is still present even if the tokens are empty
+        //   Example: jsmith:ReadWrite:true:::
+        // The first token (username) is allowed to contain embedded ":".
+        String[] parts = new String[6];
```

Comment:
Define `parts` just before the `for` loop using `new String[m.groupCount()]` to avoid hardcoding the `6`.